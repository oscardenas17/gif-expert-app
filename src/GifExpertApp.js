// rafc
import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ( {defaultCategories = []}) => {

    // const categories = ['One Punch','Samurai X','Dragon Ball'];
    //Hook de useState - 
    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);


    // const handleAdd = () =>{
    //     setcategories( [...categories, 'HunterXhunter'] );
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories= {setCategories} />
            <hr />

            {/* <button onClick={handleAdd}> Agregar</button> */}

            <ol>
                {
                    categories.map( category =>(
                        <GifGrid 
                        key = {category}
                        category= {category} />
                         ) )
                }
            </ol>
        </>
    )
}

