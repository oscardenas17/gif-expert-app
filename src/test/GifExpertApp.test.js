import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en <GifExpertApp />', () => {
    
    test('Snapshot ok', () => {

        const wrapper = shallow (<GifExpertApp /> );

        expect(wrapper).toMatchSnapshot();
    })


    test('debe de mostrar una lista da categorias', () => {

        const categories = ['DBZ','ONE']
        const wrapper = shallow (<GifExpertApp  defaultCategories={categories}/> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Gifgrid').length).toBe(categories);
    })
    
})
