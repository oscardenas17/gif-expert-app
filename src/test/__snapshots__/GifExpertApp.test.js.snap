// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Snapshot ok 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista da categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
